flink-core/src/test/java/org/apache/flink/util/MdcUtilsTest.java

```diff
@@ -50,6 +56,19 @@ class MdcUtilsTest {
     public final LoggerAuditingExtension loggerExtension =
             new LoggerAuditingExtension(MdcUtilsTest.class, DEBUG);
 
+    @Test
+    void testContextRestorationWorksWithNullContext() {
+        try (MockedStatic<MDC> mockStatic = Mockito.mockStatic(MDC.class, new CallsRealMethods())) {
+            mockStatic.when(MDC::getCopyOfContextMap).thenReturn(null);
+            mockStatic.when(() -> MDC.setContextMap(isNull()))
+                    .thenThrow(NullPointerException.class);
+
+            MdcCloseable restoreContext = MdcUtils.withContext(Collections.singletonMap("k", "v"));
+            assertThat(MDC.get("k")).isEqualTo("v");
+            assertDoesNotThrow(restoreContext::close);
```

Comment:
could we have a test for .clear() also please

Reply:
What kind of test for `.clear()` - what should I check?